src/wp-includes/webfonts-api/providers/class-wp-webfonts-google-provider.php

```diff
@@ -0,0 +1,326 @@
+				}
+				$font_families[ $webfont['font-family'] ][] = $webfont;
+			}
+
+			$font_display_groups[ $font_display ] = $font_families;
+		}
+
+		return $font_display_groups;
+	}
+
+	/**
+	 * Collects all font-weights grouped by 'normal' and 'italic' font-style.
+	 *
+	 * For example, if given the following webfonts:
+	 * ```
+	 * array(
+	 *      array(
+	 *          'font-family' => 'Source Serif Pro',
+	 *          'font-style'  => 'normal',
+	 *          'font-weight' => '200 400',
+	 *      ),
+	 *      array(
+	 *          'font-family' => 'Source Serif Pro',
+	 *          'font-style'  => 'italic',
+	 *          'font-weight' => '400 600',
+	 *      ),
+	 * )
+	 * ```
+	 * Then the returned collection would be:
+	 * ```
+	 * array(
+	 *      array( 200, 300, 400 ),
+	 *      array( 400, 500, 600 ),
+	 * )
+	 * ```
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array $webfonts Webfonts to process.
+	 * @return array[] {
+	 *    The font-weights grouped by font-style.
+	 *
+	 *    @type array $normal_weights  Individual font-weight values for 'normal' font-style.
+	 *    @type array $italic_weights  Individual font-weight values for 'italic' font-style.
+	 * }
+	 */
+	private function collect_font_weights( array $webfonts ) {
+		$normal_weights = array();
+		$italic_weights = array();
+
+		foreach ( $webfonts as $webfont ) {
+			$font_weights = $this->get_font_weights( $webfont['font-weight'] );
+			// Skip this webfont if it does not have a font-weight defined.
+			if ( empty( $font_weights ) ) {
+				continue;
+			}
+
+			// Add the individual font-weights to the end of font-style array.
+			if ( 'italic' === $webfont['font-style'] ) {
+				array_push( $italic_weights, ...$font_weights );
+			} else {
+				array_push( $normal_weights, ...$font_weights );
+			}
+		}
+
+		// Remove duplicates.
+		$normal_weights = array_unique( $normal_weights );
+		$italic_weights = array_unique( $italic_weights );
+
+		return array( $normal_weights, $italic_weights );
+	}
+
+	/**
+	 * Converts the given string of font-weight into an array of individual weight values.
+	 *
+	 * When given a single font-weight, the value is wrapped into an array.
+	 *
+	 * A range of font-weights is specified as '400 600' with the lightest value first,
+	 * a space, and then the heaviest value last.
+	 *
+	 * When given a range of font-weight values, the range is converted into individual
+	 * font-weight values. For example, a range of '400 600' is converted into
+	 * `array( 400, 500, 600 )`.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $font_weights The font-weights string.
+	 * @return array The font-weights array.
+	 */
+	private function get_font_weights( $font_weights ) {
+		$font_weights = trim( $font_weights );
+
+		// A single font-weight.
+		if ( false === strpos( $font_weights, ' ' ) ) {
+			return array( $font_weights );
+		}
+
+		// Process a range of font-weight values that are delimited by ' '.
+		$font_weights = explode( ' ', $font_weights );
```

Comment:
I have a considerable lack of knowledge on the API decisions but I wonder why on the API we decided to pass font weights as string that then we need to parse using explode instead of passing an array?
e.g: `'font-weights'=> array( '200', '900' ) instead of 'font-weight'  => '200 900',`

Reply:
cc @aristath

Reply:
@jorgefilipecosta See [this comment](https://github.com/WordPress/wordpress-develop/pull/1736#issuecomment-961189814) for part of the discussion on this

Reply:
If the user only wants to load the `200` and `900` font-weights, then the webfont variations should be loaded separately. The `200 900` value is defining a range, as per the [CSS spec for `@font-face`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face).